specs/agents/field-limits.md

```diff
@@ -0,0 +1,31 @@
+## Field limits
+
+The maximum length of metadata, transaction, span, et al fields are determined
+by the [APM Server Events Intake API schema](https://www.elastic.co/guide/en/apm/server/current/events-api.html).
+Fields that are names or identifiers of some resource of typically limited to
+1024 unicode characters.
+
+### `long_field_max_length` configuration
+
+Some APM event fields are not limited in the APM server intake API schema.
+Agents SHOULD limit the maximum length of the following fields by truncating.
+
+- `{transaction,error}.context.request.body`
+- `{transaction,span,error}.context.message.body`
```

Comment:
[suggestion] maybe expand to full field names to make this easier to read.

Reply:
Agreed. Done in commit 2deecbe304b0d2592c998ab8a8a25f2e988cfd3a.